packages/neon/neon/lib/src/blocs/unified_search.dart

```diff
@@ -0,0 +1,109 @@
+part of 'blocs.dart';
+
+abstract interface class UnifiedSearchBlocEvents {
+  void search(final String term);
+}
+
+abstract interface class UnifiedSearchBlocStates {
+  BehaviorSubject<Result<Iterable<NextcloudCoreUnifiedSearchProvider>>> get providers;
+
+  BehaviorSubject<Map<NextcloudCoreUnifiedSearchProvider, Result<NextcloudCoreUnifiedSearchResult>>> get results;
+}
+
+class UnifiedSearchBloc extends InteractiveBloc implements UnifiedSearchBlocEvents, UnifiedSearchBlocStates {
+  UnifiedSearchBloc(
+    this._requestManager,
+    this._client,
+    final Iterable<AppImplementation>? appImplementations,
+  ) : _appImplementations = appImplementations ?? [];
+
+  final RequestManager _requestManager;
+  final NextcloudClient _client;
+  final Iterable<AppImplementation> _appImplementations;
+  late final _implementedProviders =
+      _appImplementations.expand((final app) => app.unifiedSearchProviders?.toList() ?? []).toSet();
```

Comment:
```dart
      _appImplementations.expand((final app) => app.unifiedSearchProviders ?? {}).toSet();
```
should also work

Reply:
No, it expects a list to be returned

Reply:
Just checked and it does work.
Obviously the part after `??` must be changed to match the type of unifiedSearchProviders (a Set)

We would save a couple iterations this way

Reply:
Huh maybe I did something wrong. Will fix it in the next iteration